Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift

```diff
@@ -241,16 +246,34 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
   }
   
   func initLatencyPicker() {
-    let latencyPosition = Latency.allTimeValues.index(of: UserDefaults.standard.integer(forKey: kMSTransmissionIterval))
-    self.latencyPicker = MSEnumPicker<Latency>(
-    textField: latencyField,
-    allValues: Latency.allValues,
-    onChange: { index in
-      UserDefaults.standard.setValue(Latency.allTimeValues[index], forKey: kMSTransmissionIterval)
-    })
-    self.latency = Latency.allValues[latencyPosition ?? 0]
-    self.latencyField.delegate = self.latencyPicker
-    self.latencyField.text = self.latency.rawValue
-    self.latencyField.tintColor = UIColor.clear
+    transmissionIntervalField.inputAccessoryView = self.toolBarForKeyboard()
+    let interval = UserDefaults.standard.integer(forKey: kMSTransmissionIterval) != 0 ? UserDefaults.standard.integer(forKey: kMSTransmissionIterval) : 3
+    transmissionIntervalField.text = String(interval)
+    updateIntervalLabel(transmissionInterval: interval)
+  }
+  
+  func secondsToHoursMinutesSeconds(seconds : Int) -> (Int, Int, Int) {
```

Comment:
I'd go for built-in functionality. [DateComponentsFormatter](https://developer.apple.com/documentation/foundation/datecomponentsformatter)